swagger/src/main/scala/org/http4s/rho/swagger/SwaggerSupport.scala

```diff
@@ -8,28 +8,40 @@ import headers.`Content-Type`
 import org.http4s.rho.bits.PathAST.TypedPath
 import org.http4s.rho.swagger.models.{Swagger, Info}
 
-import shapeless.HNil
+import shapeless._
 
 object SwaggerSupport {
+
+  /**
+   * Create a RhoMiddleware adding a route to get the Swagger json file
+   * representing the full API
+   */
   def apply(
     swaggerFormats: SwaggerFormats = DefaultSwaggerFormats,
     apiPath: TypedPath[HNil] = "swagger.json",
     apiInfo: Info = Info(title = "My API", version = "1.0.0")): RhoMiddleware = { routes =>
-    val sb = new SwaggerModelsBuilder(swaggerFormats)
-    var swaggerSpec: Swagger = Swagger()
 
-    val swaggerRoutes = new RhoService {
-      lazy val swaggerResponse =
-      Ok(Json.mapper().writeValueAsString(swaggerSpec.toJModel))
-        .putHeaders(`Content-Type`(MediaType.`application/json`))
+    lazy val swaggerSpec: Swagger =
+      createSwagger(swaggerFormats, apiPath, apiInfo)(routes ++ swaggerRoutes)
 
-      "Swagger documentation" ** GET / apiPath |>> { () => swaggerResponse }
-    }.getRoutes()
+    lazy val swaggerRoutes: Seq[RhoRoute[_ <: HList]] = new RhoService {
+      "Swagger documentation" ** GET / apiPath |>> { () => Ok(Json.mapper().writeValueAsString(swaggerSpec.toJModel))
```

Comment:
The one thing I liked about the old behavior is that the swagger spec was memorized. I think that behavior can be preserved.